--- Node.pm	2020-10-30 00:39:52.000000000 +0100
+++ Node.pm.new	2021-04-07 16:38:43.000000000 +0200
@@ -5,6 +5,7 @@
 
 use NetAddr::MAC;
 use App::Netdisco::Util::Permission qw/check_acl_no check_acl_only/;
+use App::Netdisco::Util::DNS 'ipv4_from_hostname';
 
 use base 'Exporter';
 our @EXPORT = ();
@@ -101,9 +102,8 @@
 
   # VRRP
   if ($mac->is_vrrp) {
-      debug sprintf ' [%s] check_mac - VRRP mac [%s] - skipping',
-        $devip, $node;
-      return 0;
+      debug sprintf ' [%s] check_mac - VRRP mac [%s] - still including', $devip, $node;
+      #return 0;
   }
 
   # HSRP
@@ -165,7 +165,7 @@
 =cut
 
 sub store_arp {
-  my ($hash_ref, $now) = @_;
+  my ($hash_ref, $now, $device_ip) = @_;
   $now ||= 'now()';
   my $ip   = $hash_ref->{'ip'};
   my $mac  = NetAddr::MAC->new(mac => ($hash_ref->{'node'} || ''));
@@ -193,6 +193,27 @@
         for => 'update',
       });
   });
+
+
+    if (setting('arpnip_track_l3device')){
+
+        schema('netdisco')->resultset('NodeIpL3device')
+          ->update_or_create(
+          {
+            node_mac => $mac->as_ieee,
+            node_ip => $ip,
+            device_ip => $device_ip,
+            time_last => \$now,
+          },
+          {
+            key => 'primary',
+            for => 'update',
+          });
+
+        }
+
+
+
 }
 
 1;
